src/pages/AuthorizationPage.tsx

```diff
@@ -0,0 +1,132 @@
+import { FC, useState, FormEvent } from 'react';
+import { User } from '../types/User';
+import { ErrorMessage } from '../types/ErrorMessage';
+import { ErrorNotification } from '../components/ErrorNotification';
+import { createUser, getUser } from '../api/user';
+
+type Props = {
+  setUser: (newValue: User | null) => void;
+};
+
+export const AuthorizationPage: FC<Props> = ({ setUser }) => {
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [isUserNotFind, setIsUserNotFind] = useState(false);
+  const [isError, setIsError] = useState<ErrorMessage>(ErrorMessage.None);
+
+  const fetchUser = async () => {
+    try {
+      const getData = await getUser(email);
+
+      if (!getData.length) {
+        setIsUserNotFind(true);
+        setIsError(ErrorMessage.LoadUser);
+      }
+
+      const user = getData.find(data => {
+        return data.email === email && data.name !== '';
+      });
+
+      if (user) {
+        setUser(user);
+      }
+    } catch (error) {
+      setIsError(ErrorMessage.Login);
+    }
+  };
+
+  const addUser = async () => {
+    const newUser = {
+      email,
+      name,
+    };
+
+    try {
+      await createUser(newUser as User);
+    } catch {
+      setIsError(ErrorMessage.Load);
+    }
+  };
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!isUserNotFind) {
+      fetchUser();
+    } else {
+      await addUser();
+      await fetchUser();
+      setIsUserNotFind(false);
+    }
+  };
+
+  return (
+    <section className="section">
+      <form
+        method="post"
+        onSubmit={(e) => handleSubmit(e)}
```

Comment:
work same but less callbacks
```suggestion
        onSubmit={handleSubmit}
```